### PR TITLE
Properly install Pattern library for documentation build

### LINKED ISSUE #2534: "Pattern library is not installed" ... but it is (perhaps because it is called pattern3)
#### Problem description

I am trying to do a gensim tutorial project. I am attempting to lemmatize and getting this error:
ImportError: Pattern library is not installed. Pattern library is needed in order to use lemmatize function

After searching the web, I installed pattern3, I still get the same error. I tried to do 'import pattern3 as pattern' but that doesn't help. Obviously I'm on python 3 (3.6)

#### Steps/code/corpus to reproduce

 import gensim
 from gensim.utils import simple_preprocess, lemmatize
 import pattern3

lemmatize("hello")

#### Versions

Please provide the output of:

```python
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("NumPy", numpy.__version__)
import scipy; print("SciPy", scipy.__version__)
import gensim; print("gensim", gensim.__version__)
from gensim.models import word2vec;print("FAST_VERSION", word2vec.FAST_VERSION)
```

Darwin-15.6.0-x86_64-i386-64bit
Python 3.6.8 |Anaconda, Inc.| (default, Dec 29 2018, 19:04:46) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
NumPy 1.16.4
SciPy 1.2.1
gensim 3.7.3
FAST_VERSION 1

#### Additional info

`from pattern3.web import Twitter`
this works fine

`from pattern.web import Twitter`
this does not (even after I do `import pattern3 as pattern`)

### PR DESCRIPTION
**Pull request by Hiyorimi**
Fixed dependency in the only place I could find. 

Fix #2534

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -276,13 +276,13 @@ def finalize_options(self):
...
-    linux_testenv.extend(['nmslib'])    
-    
+    linux_testenv.extend(['nmslib'])
+
 docs_testenv = linux_testenv + distributed_env + [
     'sphinx',
     'sphinxcontrib-napoleon',
     'plotly',
-    'pattern <= 2.6',
+    'Pattern >= 3.6',
```
Comment (mpenkov) [inline setup.py]: Is there a reason why we need to specify the version here?

Comment (Hiyorimi) [reply setup.py]: To get version with python3 support.

Comment (mpenkov) [reply setup.py]: Could you please mention that in a code comment? Someone is likely to ask the same question later on.

Comment (Hiyorimi) [reply setup.py]: Pushed

**Comment by mpenkov** [comment]
Looks good to me. Congrats on your first contribution to gensim :1st_place_medal: !

**Comment by Hiyorimi** [comment]
Thx!

**Pull request merged**